msk-lambda-schema-avro-python-sam/kafka_event_producer_function/app.py

```diff
@@ -0,0 +1,272 @@
+            value_serializer=lambda x: x,  # Raw bytes - AVRO data is already serialized
+            acks='all',
+            retries=3,
+            max_block_ms=120000,
+            request_timeout_ms=60000,
+        )
+        
+        logger.info("Starting message production", extra={
+            "topic": kafka_topic,
+            "message_count": message_count,
+            "schema_version_id": str(schema_version_id)
+        })
+        
+        # Track zip code distribution for filtering demo
+        zip_1000_count = 0
+        zip_2000_count = 0
+        
+        # Send messages
+        for i in range(message_count):
+            contact = create_sample_contact(i)
+            message_key = f"contact-{i+1}"
+            
+            # Track zip code distribution
+            if contact['zip'].startswith('1000'):
+                zip_1000_count += 1
+            elif contact['zip'].startswith('2000'):
+                zip_2000_count += 1
+            
+            # Serialize and send message
+            avro_message = serialize_avro_message(contact, schema_version_id)
+            future = producer.send(kafka_topic, key=message_key, value=avro_message)
+            record_metadata = future.get(timeout=60)
+            
+            logger.info("Message sent successfully", extra={
+                "message_number": i+1,
+                "message_key": message_key,
+                "partition": record_metadata.partition,
+                "offset": record_metadata.offset,
+                "message_size": len(avro_message)
+            })
+            
+            # Add metrics
+            metrics.add_metric(name="AvroMessagesSent", unit=MetricUnit.Count, value=1)
+            metrics.add_metric(name="AvroMessageSize", unit=MetricUnit.Bytes, value=len(avro_message))
+        
+        # Add distribution metrics for filtering demo
+        metrics.add_metric(name="AvroMessages1000Prefix", unit=MetricUnit.Count, value=zip_1000_count)
+        metrics.add_metric(name="AvroMessages2000Prefix", unit=MetricUnit.Count, value=zip_2000_count)
+        
+        # Close producer
+        producer.close()
+        
+        success_message = (
+            f"Successfully sent {message_count} AVRO messages to Kafka topic: {kafka_topic} "
+            f"using schema {schema_name} (version {schema_version_id}) from registry {registry_name} "
+            f"(Zip codes: {zip_1000_count} with prefix 1000, {zip_2000_count} with prefix 2000)"
+        )
+        
+        logger.info("MSK AVRO Producer Lambda completed successfully", extra={
+            "success_message": success_message,
+            "total_messages_sent": message_count,
+            "schema_version_id": str(schema_version_id)
+        })
+        
+        return success_message
+        
+    except Exception as e:
+        logger.exception("Error in lambda_handler", extra={
+            "error": str(e),
+            "error_type": type(e).__name__
+        })
```

Comment:
logger.exception already propagate error type and name.

```suggestion
        logger.exception("Error in lambda_handler")
```